Assets/FeatureVoid/VoidMechaFollower.cs

```diff
@@ -0,0 +1,16 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+public class VoidMechaFollower : MonoBehaviour {
+
+    public GameObject mecha;
+	
+	// Update is called once per frame
+	void Update () {
+        // The void collider stays under the mecha and follows the player
+        // along the z and x axis.
+        transform.position = new Vector3(mecha.transform.position.x, transform.position.y, mecha.transform.position.z);
```

Comment:
Since the mecha will "rotate" in the cylinder, you should also set the rotation of the mecha.

Reply:
I believe I can't easily fix this for the moment as the mecha is a ball and moving sideways affects its z-axis rotation.